packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -370,18 +383,7 @@ export default createRule<Options, MessageId>({
 
       const type = getConstrainedTypeAtLocation(services, expression);
 
-      // Conditional is always necessary if it involves:
-      //    `any` or `unknown` or a naked type variable
-      if (
-        tsutils
-          .unionTypeParts(type)
-          .some(
-            part =>
-              isTypeAnyType(part) ||
-              isTypeUnknownType(part) ||
-              isTypeFlagSet(part, ts.TypeFlags.TypeVariable),
-          )
-      ) {
```

Comment:
[Praise] Heh, yeah, this was getting a little hard to read inline even before...